windows-server-containers-preview/azuredeploy.json

```diff
@@ -0,0 +1,271 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+
+     "VMName": {
+      "type": "string",
+      "metadata": {
+        "description": "This name will also be used for the network security group, storage, storage container, virtual network, network card, subnet and public IP address name."
```

Comment:
I suggest two approaches here
1. For storage account name, make it a parameter so that the user can input a globally unique name
2. For others, make them a variable and follow this naming scheme - <template scenario><resource e.g. nsg, pubip, vnet> for better identification when querying resource group or resources in Portal